services/orchest-api/app/app/models.py

```diff
@@ -22,6 +25,19 @@ class BaseModel(db.Model):
     def as_dict(self):
         return {c.name: getattr(self, c.name) for c in self.__table__.columns}
 
+    @classmethod
+    def keep_column_entries(cls, data: Dict[str, Any]) -> Dict[str, Any]:
+        """Remove entries not related to the model columns from a dict.
+
+        Can be used to sanitize a status update.
+        """
+        data = copy.deepcopy(data)
+        columns = [c.name for c in cls.__table__.columns]
+        for key in list(data.keys()):
+            if key not in columns:
+                data.pop(key)
+        return data
+
```

Comment:
The following should be a little faster, since a deepcopy is only created for entries that are a column. (Btw, you can also just do `for key in data` as that is short for `for key in data.keys()` -- no conversion to `list` is needed.)
```suggestion
        ans = {}
        columns = [c.name for c in cls.__table__.columns]
        for key, value in data.items():
            if key in columns:
                ans[key] = copy.deepcopy(value)
        return ans
```